exchanges/deribit/deribit_websocket.go

```diff
@@ -0,0 +1,1212 @@
+	priceRankingChannel                    = "deribit_price_ranking"
+	priceStatisticsChannel                 = "deribit_price_statistics"
+	volatilityIndexChannel                 = "deribit_volatility_index"
+	estimatedExpirationPriceChannel        = "estimated_expiration_price"
+	incrementalTickerChannel               = "incremental_ticker"
+	instrumentStateChannel                 = "instrument.state"
+	markPriceOptionsChannel                = "markprice.options"
+	perpetualChannel                       = "perpetual."
+	platformStateChannel                   = "platform_state"
+	platformStatePublicMethodsStateChannel = "platform_state.public_methods_state"
+	quoteChannel                           = "quote"
+	requestForQuoteChannel                 = "rfq"
+	tickerChannel                          = "ticker."
+	tradesChannel                          = "trades."
+	tradesWithKindChannel                  = "trades"
+
+	// private websocket channels
+	userAccessLogChannel                             = "user.access_log"
+	userChangesInstrumentsChannel                    = "user.changes."
+	userChangesCurrencyChannel                       = "user.changes"
+	userLockChannel                                  = "user.lock"
+	userMMPTriggerChannel                            = "user.mmp_trigger"
+	rawUserOrdersChannel                             = "user.orders.%s.raw"
+	userOrdersWithIntervalChannel                    = "user.orders."
+	rawUsersOrdersKindCurrencyChannel                = "user.orders.%s.%s.raw"
+	rawUsersOrdersWithKindCurrencyAndIntervalChannel = "user.orders"
+	userPortfolioChannel                             = "user.portfolio"
+	userTradesChannelByInstrument                    = "user.trades."
+	userTradesByKindCurrencyAndIntervalChannel       = "user.trades"
+)
+
+var defaultSubscriptions = []string{
+	chartTradesChannel, // chart trades channel to fetch candlestick data.
+	orderbookChannel,
+	tickerChannel,
+	tradesWithKindChannel,
+}
+
+var indexENUMS = []string{"ada_usd", "algo_usd", "avax_usd", "bch_usd", "bnb_usd", "btc_usd", "doge_usd", "dot_usd", "eth_usd", "link_usd", "ltc_usd", "luna_usd", "matic_usd", "near_usd", "shib_usd", "sol_usd", "trx_usd", "uni_usd", "usdc_usd", "xrp_usd", "ada_usdc", "bch_usdc", "algo_usdc", "avax_usdc", "btc_usdc", "doge_usdc", "dot_usdc", "bch_usdc", "bnb_usdc", "eth_usdc", "link_usdc", "ltc_usdc", "luna_usdc", "matic_usdc", "near_usdc", "shib_usdc", "sol_usdc", "trx_usdc", "uni_usdc", "xrp_usdc", "btcdvol_usdc", "ethdvol_usdc"}
+
+var (
+	pingMessage = WsSubscriptionInput{
+		ID:             2,
+		JSONRPCVersion: rpcVersion,
+		Method:         "public/test",
+		Params:         map[string][]string{},
+	}
+	setHeartBeatMessage = wsInput{
+		ID:             1,
+		JSONRPCVersion: rpcVersion,
+		Method:         "public/set_heartbeat",
+		Params: map[string]interface{}{
+			"interval": 15,
+		},
+	}
+)
+
+// WsConnect starts a new connection with the websocket API
+func (d *Deribit) WsConnect() error {
```

Comment:
[ERROR] | EXCHANGE | 01/06/2023 10:00:54 | subscription to channel chart.trades.LTC-USDC-PERPETUAL.1 was not successful

websocket enabled - auth not enabled